### PR TITLE
WIP - V1.15 compatibility

### PR DESCRIPTION
**Pull request by jimconner**
This PR contains changes that allow HouseRules to compile against Demeo v1.15.158895. 

Mods all appear to load OK, but currently hand controllers do not work as input when MelonLoader is installed. As such it has not be possible to functionally test anything.

### CODE REVIEW

**Comment by jimconner** [comment]
A working melonloader release is now avaialbe as a build-artifact... https://github.com/LavaGang/MelonLoader/actions/runs/2394642822 ... meaning that it is now possible to test this PR.

File: RoomFinder/RoomFinder.csproj [hunk 1]
```diff
@@ -1,83 +1,86 @@
...
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="0Harmony">
+      <HintPath>$(DemeoVrDir)\MelonLoader\0Harmony.dll</HintPath>
+    </Reference>
+    <Reference Include="Assembly-CSharp">
+      <HintPath>$(DemeoVrDir)\demeo_Data\Managed\Assembly-CSharp.dll</HintPath>
+    </Reference>
+    <Reference Include="BowserCore">
+      <HintPath>D:\Occulus\Software\resolution-games-demeo\Demeo VR\demeo_Data\Managed\BowserCore.dll</HintPath>
```
Comment (orendain) [inline RoomFinder/RoomFinder.csproj:42]: Could we change this to one based on the variable path (i.e., `$(DemeoVrDir)\...`)?

File: Common/Common.csproj [hunk 2]
```diff
@@ -1,66 +1,70 @@
...
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="0Harmony">
+      <HintPath>$(DemeoVrDir)\MelonLoader\0Harmony.dll</HintPath>
+    </Reference>
+    <Reference Include="BowserCore, Version=0.0.0.0, Culture=neutral, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>D:\Occulus\Software\resolution-games-demeo\Demeo VR\demeo_Data\Managed\BowserCore.dll</HintPath>
```
Comment (orendain) [inline Common/Common.csproj:40]: Here too (and, if possible, without additional properties (e.g., culture=, processorArch...`).

File: HouseRules_Configuration/HouseRules_Configuration.csproj [hunk 3]
```diff
@@ -1,86 +1,89 @@
...
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="0Harmony">
+      <HintPath>$(DemeoVrDir)\MelonLoader\0Harmony.dll</HintPath>
+    </Reference>
+    <Reference Include="Assembly-CSharp">
+      <HintPath>$(DemeoVrDir)\demeo_Data\Managed\Assembly-CSharp.dll</HintPath>
+    </Reference>
+    <Reference Include="BowserCore">
+      <HintPath>D:\Occulus\Software\resolution-games-demeo\Demeo VR\demeo_Data\Managed\BowserCore.dll</HintPath>
```
Comment (orendain) [inline HouseRules_Configuration/HouseRules_Configuration.csproj:42]: Here too.

File: HouseRules_Core/HouseRules_Core.csproj [hunk 4]
```diff
@@ -1,79 +1,82 @@
...
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="0Harmony">
+      <HintPath>$(DemeoVrDir)\MelonLoader\0Harmony.dll</HintPath>
+    </Reference>
+    <Reference Include="Assembly-CSharp">
+      <HintPath>$(DemeoVrDir)\demeo_Data\Managed\Assembly-CSharp.dll</HintPath>
+    </Reference>
+    <Reference Include="BowserCore">
+      <HintPath>D:\Occulus\Software\resolution-games-demeo\Demeo VR\demeo_Data\Managed\BowserCore.dll</HintPath>
```
Comment (orendain) [inline HouseRules_Core/HouseRules_Core.csproj:42]: One more!

File: HouseRules_Essentials/Rules/StartCardsModifiedRule.cs [hunk 5]
```diff
@@ -26,7 +26,6 @@ public struct CardConfig
 
         public StartCardsModifiedRule(Dictionary<BoardPieceId, List<CardConfig>> heroStartCards)
         {
-            ValidateCards(heroStartCards);
```
Comment (orendain) [inline HouseRules_Essentials/Rules/StartCardsModifiedRule.cs:29]: Niiiiiiice!

File: HouseRules_Core/HangoutsGameRacer.cs [hunk 6]
```diff
@@ -185,7 +185,7 @@ private static void PrepareToLeaveHangouts(GroupLaunchTable groupLaunchTable)
 
             // Recreating `GameStateHobbyShop.ExitBowser`
             var teleport = Traverse.Create(_gameStateHobbyShop).Field<Teleport>("teleport").Value;
-            teleport.SetTeleportationEnabled(enabled: false);
+            teleport.UpdateState(true, false);
```
Comment (orendain) [inline HouseRules_Core/HangoutsGameRacer.cs:188]: nit: for clarity on what booleans do, could we inline parameter names?  (e.g., ...UpdateState(paramName1: true, paramName2: false) (don't have access to the code ATM so don't know what these are :p).

**Review by orendain — APPROVED** [review]
LGTM, thanks @jimconner!  Just minor notes.  Not on a machine I can build on ATM, so leaving merging to you.

File: RoomFinder/RoomFinder.csproj (hunk 1 above)
Comment (jimconner) [reply RoomFinder/RoomFinder.csproj]: Sorry - my bad for letting visual studio add these for me automatically.. I should have gone back and tidied up afterwards.

**Pull request merged**